drivers/flash/flash_mcux_c40.c

```diff
@@ -0,0 +1,381 @@
+	}
+
+	key = irq_lock();
+
+	barrier_dsync_fence_full();
+
+	z_barrier_isync_fence_full();
+
+	st = FLASH_Erase(&data->cfg, (uint32_t)(cfg->base + (uintptr_t)off), (uint32_t)len,
+			 kFLASH_ApiEraseKey);
+
+	barrier_dsync_fence_full();
+
+	irq_unlock(key);
+
+	sys_cache_data_invd_range((void *)(cfg->base + (uintptr_t)off), len);
+	return mcux_to_errno(st);
+}
+
+static const struct flash_parameters *mcux_c40_get_parameters(const struct device *dev)
+{
+	const struct mcux_c40_cfg *cfg = dev->config;
+
+	return cfg->params;
+}
+
+#ifdef CONFIG_FLASH_PAGE_LAYOUT
+static void mcux_c40_pages_layout(const struct device *dev,
+				  const struct flash_pages_layout **layout, size_t *layout_size)
+{
+	const struct mcux_c40_cfg *cfg = dev->config;
+	static struct flash_pages_layout l;
+
+	l.pages_count = cfg->size / cfg->erase_block;
+	l.pages_size = cfg->erase_block;
+
+	*layout = &l;
+	*layout_size = 1;
+}
+#endif
+
+/* Optional “lock policy” executed at init (opt-in via Kconfig) */
+#if defined(CONFIG_SOC_FLASH_MCUX_C40_APPLY_PROTECTION)
+/* We place this TU into .ram_text_reloc via CMake, so no special attribute is needed. */
+static __attribute__((noinline)) status_t c40_apply_protection(flash_config_t *fcfg,
+							       uint32_t flash_base,
+							       uint32_t total_sz, uint32_t erase_sz,
+							       const struct prot_range *pr,
+							       size_t npr)
+{
+	uint32_t off;
+	uint32_t abs;
+	unsigned int key;
+	size_t i;
+	bool lock_it;
+
+	for (off = 0; off < total_sz; off += erase_sz) {
+		status_t ps;
+
+		lock_it = false;
+
+		for (i = 0; i < npr; i++) {
+			if (intersects(off, erase_sz, pr[i].off, pr[i].len)) {
+				lock_it = true;
+				break;
+			}
+		}
+		abs = flash_base + off;
+
+		ps = FLASH_GetSectorProtection(fcfg, abs);
+		if (lock_it) {
+			status_t s;
+
+			if (ps != kStatus_FLASH_SectorLocked) {
+				key = irq_lock();
+
+				/* Didn't use ISB here since there is no
+				 * instruction side stream change
+				 */
+				barrier_dsync_fence_full();
+
+				s = FLASH_SetSectorProtection(fcfg, abs, true);
+
+				barrier_dsync_fence_full();
+
+				irq_unlock(key);
+				if (s != kStatus_FLASH_Success) {
+					return s;
+				}
+			}
+		} else {
+			status_t s;
+
+			if (ps != kStatus_FLASH_SectorUnlocked) {
+				key = irq_lock();
+
+				barrier_dsync_fence_full();
+
+				s = FLASH_SetSectorProtection(fcfg, abs, false);
+
+				barrier_dsync_fence_full();
+
+				irq_unlock(key);
+				if (s != kStatus_FLASH_Success) {
+					return s;
+				}
+			}
+		}
+	}
+	return kStatus_FLASH_Success;
+}
+#endif /* CONFIG_SOC_FLASH_MCUX_C40_APPLY_PROTECTION */
+
+static int mcux_c40_init(const struct device *dev)
+{
+	const struct mcux_c40_cfg *cfg = dev->config;
+	struct mcux_c40_data *data = dev->data;
+	status_t st;
+
+	st = FLASH_Init(&data->cfg);
+	if (st != kStatus_FLASH_Success) {
+		LOG_ERR("FLASH_Init failed: %d", (int)st);
+		return mcux_to_errno(st);
+	}
+
+	LOG_INF("C40 flash: base=0x%lx size=0x%lx erase=0x%lx write=0x%lx",
```

Comment:
```suggestion
	LOG_DBG("C40 flash: base=0x%lx size=0x%lx erase=0x%lx write=0x%lx",
```
Apply for all occurrences.